std/algorithm/iteration.d

```diff
@@ -2415,6 +2415,15 @@ if (isInputRange!RoR && isInputRange!(ElementType!RoR))
                 return copy;
             }
         }
+
+        static if (hasAssignableElements!(ElementType!RoR))
+        {
+            @property void front(ElementType!(ElementType!RoR) element)
```

Comment:
Maybe an overload in `hasLvalueElements` with a `ref` parameter would be appropriate?